File: examples/rust/bounded-loop/BoundedLoop.runar.rs
use runar::prelude::*;

#[runar::contract]
struct BoundedLoop {
    #[readonly]
    expected_sum: Int,
}

impl BoundedLoop {
    pub fn verify(&self, start: Int) {
        let mut sum: Int = 0;
        for i in 0..5 {
            sum = sum + start + i;
        }
        assert!(sum == self.expected_sum);
    }
}

File: examples/go/bounded-loop/BoundedLoop.runar.go
package contract

import runar "github.com/icellan/runar/packages/runar-go"

type BoundedLoop struct {
	runar.SmartContract
	ExpectedSum runar.Int `runar:"readonly"`
}

func (c *BoundedLoop) Verify(start runar.Int) {
	sum := runar.Int(0)
	for i := runar.Int(0); i < 5; i++ {
		sum = sum + start + i
	}
	runar.Assert(sum == c.ExpectedSum)
}
